content/introduction/how-ipfs-works.md

```diff
@@ -0,0 +1,82 @@
+
+<!-- <img alt="IPFS Stack" src="../assets/ipfs_stack.png" width="200px" /> -->
+
+## There are three important things to understand about IPFS
+
+Let’s first dive in to _content addressing_ and how that content is _linked together_. This “middle” part of the IPFS stack is what connects the ecosystem together; everything is built on being able to find content via linked, unique identifiers.
+
+### 1 \\ Content addressing and linked data
+
+<!-- <img alt="IPFS Stack - Data Structures" src="../assets/ipfs_stack-data.png" width="200px" /> -->
+
+IPFS uses _content addressing_ to identify content by what’s in it, rather than by where it’s located. Looking for an item by content is actually something you do all the time. For example, when you look for a book in the library, you ask for it by the title; that’s content addressing because you’re asking for **what** it is. If you were using location addressing to find that book, you’d ask for it by **where** it is: “I want the book that’s on the second floor, first stack, third shelf from the bottom, four books from the left.” If someone moved that book, you’d be out of luck!
+
+It’s the same on the internet and on your computer. Right now, content is found by location, such as…
+
+- `https://en.wikipedia.org/wiki/Aardvark`
+- `/Users/Alice/Documents/term_paper.doc`
+- `C:\Users\Joe\My Documents\project_sprint_presentation.ppt`
+
+By contast, every piece of content that uses the IPFS protocol has a [*content identifier*]({{<relref "guides/concepts/cid.md">}}), or CID, that is its *hash*. The hash is unique to the content that it came from, even though it may look short compared to the original content. _If hashes are new to you, check out [the concept guide on hashes]({{<relref "guides/concepts/hashes.md">}}) for a good introduction._
+
+Content addressing through hashes has become a widely-used means of connecting data in distributed systems, from the commits that back your code to the blockchains that run cryptocurrencies. However, the underlying data structures in these systems are not necessarily interoperable.
+
+This is where the [IPLD project](https://ipld.io/) comes in. **Hashes identify content, and IPLD links that content together**. Since different distributed systems structure their data in different ways, IPLD provides libraries for combining pluggable modules (parsers for each possible type of IPLD node) to resolve a path, selector, or query across many linked nodes (allowing you explore data regardless of the underlying protocol). IPLD really refers to any content addressable data with links to other content addressable data: “Oh you use Git-style, no worries, I can follow those links. Oh you use Ethereum, I got you, I can follow those links too!”
```

Comment:
In retrospective, IPLD may be over-explained here. I see IPLD is about interoperability of Merkle-DAGS. It's similar to Multiformats in that sense (but we don't explain multiformats). As long as people understand the Merkle-DAG, they don't need to know much about IPLD, only that is a common language to understand different Merkle-DAG serialization formats.

Reply:
You have a point on calling IPLD a linking system. I've changed this paragraph to explain IPLD as a translation layer instead in https://github.com/ipfs/docs/pull/170/commits/ce5b46ac4b8a19e9caa91bf2c8eb9665ba294836 . 

It is important to keep the reference to IPLD is this doc because one of our ecosystem's major communication problems is people not understanding how all our projects fit together. Explaining that in a relatively straightforward way is important.